server/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -544,20 +546,20 @@ public Clusters(StreamInput in) throws IOException {
             if (in.getTransportVersion().onOrAfter(TransportVersions.V_8_500_053)) {
                 List<Cluster> clusterList = in.readCollectionAsList(Cluster::new);
                 if (clusterList.isEmpty()) {
-                    this.clusterInfo = Collections.emptyMap();
+                    this.clusterInfo = ConcurrentCollections.newConcurrentMap();
```

Comment:
Can we still leave it as `Collections.emptyMap` here or does that hit some problem?

The value of emptyMap is that it is immutable, so we can guarantee that no one ever tries to use it when local-only searches are happening.